tests/integration/test_types_integration.py

```diff
@@ -200,6 +201,119 @@ def test_digest(trino_connection):
         .execute()
 
 
+def test_date(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST(null AS DATE)", python=None) \
+        .add_field(sql="DATE '2001-08-22'", python=datetime.date(2001, 8, 22)) \
+        .add_field(sql="DATE '02001-08-22'", python=datetime.date(2001, 8, 22)) \
```

Comment:
🤯 It seems Trino allows this for some reason. @findepi Do you know why?

Reply:
DATE literals (and varchar cast to date) allow zero padding, surrounding whitespace and year sign (+ or -).
It's however unrelated to the client. 
Since `DATE '2001-08-22'` and `DATE '02001-08-22'` is just two ways of representing the same value, py client tests don't (and shouldn't) test both.

Reply:
ping @lpoulain - let's not test engine level functionality in this test. i.e. the conversion from `02001` to `2001` happens inside Trino so the client shouldn't care at all how or whether it works or not.

It might be more useful to test dates before `1582` (when Julian to Gregorian calendar switch happenned).

- `0001-01-01`
- `1582-10-04` (before the switch)
- `1582-10-05` (beginning of switch)
- `1582-10-14` (end of switch)

Reply:
Done

Reply:
reminder to remove `DATE '02001-08-22'` from test.

Reply:
Done